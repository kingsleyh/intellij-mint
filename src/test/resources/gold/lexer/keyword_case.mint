module CoachGenerator {

  fun generate (days : Number, workoutHistory : Array(WorkoutsResult), calibrationResult : CalibrationResult) : Array(Workouts) {
    case (days) {
      1 => generate1(workoutHistory, calibrationResult)
      2 => generate2(workoutHistory, calibrationResult)
      3 => generate3(workoutHistory, calibrationResult)
      4 => generate4(workoutHistory, calibrationResult)
      5 => generate5(workoutHistory, calibrationResult)
      6 => generate6(workoutHistory, calibrationResult)
      7 => generate7(workoutHistory, calibrationResult)
      => generate3(workoutHistory, calibrationResult)
    }
  }

}