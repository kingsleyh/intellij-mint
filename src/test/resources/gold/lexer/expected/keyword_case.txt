module ('module')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('CoachGenerator')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n\n  ')
fun ('fun')
WHITE_SPACE (' ')
LOWER_IDENT ('generate')
WHITE_SPACE (' ')
( ('(')
LOWER_IDENT ('days')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Number')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('workoutHistory')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Array')
( ('(')
CAPITALISED_IDENT ('WorkoutsResult')
) (')')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('CalibrationResult')
) (')')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
CAPITALISED_IDENT ('Array')
( ('(')
CAPITALISED_IDENT ('Workouts')
) (')')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
case ('case')
WHITE_SPACE (' ')
( ('(')
LOWER_IDENT ('days')
) (')')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('1')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate1')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('2')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate2')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('3')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate3')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('4')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate4')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('5')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate5')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('6')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate6')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
NUMBER_LITERAL ('7')
WHITE_SPACE (' ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate7')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n      ')
=> ('=>')
WHITE_SPACE (' ')
LOWER_IDENT ('generate3')
( ('(')
LOWER_IDENT ('workoutHistory')
, (',')
WHITE_SPACE (' ')
LOWER_IDENT ('calibrationResult')
) (')')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n')
} ('}')